include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -235,10 +235,10 @@ struct __parallel_transform_reduce_work_group_kernel_submitter<_Tp, _Commutative
                                                                __internal::__optional_kernel_name<_KernelName...>>
 {
     template <typename _ExecutionPolicy, typename _Size, typename _ReduceOp, typename _InitType>
-    auto
-    operator()(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec, sycl::event& __reduce_event,
+    __future<sycl::event, __result_and_scratch_storage<_ExecutionPolicy, _Tp>>
+    operator()(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec, sycl::event&& __reduce_event,
                const _Size __n, const _Size __work_group_size, const _Size __iters_per_work_item, _ReduceOp __reduce_op,
-               _InitType __init, const __result_and_scratch_storage<_ExecutionPolicy, _Tp>& __scratch_container) const
+               _InitType __init, __result_and_scratch_storage<_ExecutionPolicy, _Tp>&& __scratch_container) const
```

Comment:
I'd recommend using a `const sycl::event&` since you are no longer reusing and writing to it.

I'm not sure about using the rvalue ref here for the scratch space.  I think a non-const lvalue reference is maybe what I would prefer, but since we are returning it back, maybe this is better... 
I don't like it as a const lvalue ref, technically it only calls const methods, but this kernel definitely modifies its contents.

Reply:
Restored to `sycl::event&` as it was before my changes...